### DISCUSSION TITLE
如何显示详细的编译错误

### DISCUSSION THREAD

**Discussion by hqx871** [apache/doris#11090] [A - General / Q&A] [answered]
在column_predicate.h上加了个虚方法，编译失败，但是没有显示具体原因。请问如何显示具体是编译问题？
```
 virtual bool simplify(const BloomFilter* bf, ColumnPredicate* new_predicate) const {
        return false;
    }
```
./build.sh --be报错如下：
[ 61%] Linking CXX static library libOlap.a
[ 69%] Built target Olap
Makefile:148: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by hqx871** [comment, marked as answer]
/root/doris-0.14.0/be/src/gutil/hash/hash.h:299:8: note: previous definition of ‘struct __gnu_cxx::hash<std::pair<_T1, _T2> >’
  299 | struct hash<pair<First, Second>> {
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/local/doris_thirdparty/installed/include/butil/intrusive_ptr.hpp:29,
                 from /usr/local/doris_thirdparty/installed/include/brpc/channel.h:29,
                 from /root/doris-0.14.0/be/src/service/brpc.h:48,
                 from /root/doris-0.14.0/be/src/util/brpc_stub_cache.h:25,
                 from /root/doris-0.14.0/be/src/runtime/exec_env_init.cpp:56:
/usr/local/doris_thirdparty/installed/include/butil/containers/hash_tables.h:262:8: error: redefinition of ‘struct __gnu_cxx::hash<T*>’
  262 | struct hash<Type*> {
      |        ^~~~~~~~~~~
In file included from /root/doris-0.14.0/be/src/gutil/strings/stringpiece.h:124,
                 from /root/doris-0.14.0/be/src/gutil/strings/substitute.h:10,
                 from /root/doris-0.14.0/be/src/olap/rowset/segment_v2/bloom_filter.h:27,
                 from /root/doris-0.14.0/be/src/olap/column_predicate.h:25,
                 from /root/doris-0.14.0/be/src/olap/rowset/rowset_reader_context.h:21,
                 from /root/doris-0.14.0/be/src/olap/rowset/rowset_reader.h:25,
                 from /root/doris-0.14.0/be/src/olap/rowset/beta_rowset.h:25,
                 from /root/doris-0.14.0/be/src/olap/segment_loader.h:27,
                 from /root/doris-0.14.0/be/src/runtime/exec_env_init.cpp:29:
/root/doris-0.14.0/be/src/gutil/hash/hash.h:246:8: note: previous definition of ‘struct __gnu_cxx::hash<T*>’

**Reply by dataalive** [reply]
这个是为了测试还是？如果是测试的话在最新的代码里这么试试康康报错栈？

**Reply by hqx871** [reply]
已经解决了，谢谢。
